src/Backend/WASMWrapper.js

```diff
@@ -79,10 +80,56 @@ export default class WASMWrapper {
         return this.getNames(untyped, 'dedx_get_material_name')
     }
 
-    getTrace(program, ion, material) {
-        return {
-            x: [10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
-            y: Array.from(new Array(10), _ => Math.random() * 100)
-        }
+    async getTrace(program, ion, material, method, plot_using){
+        if(method === 0)
+            return await this.getTraceByInterval(program,ion,material,plot_using)
+        else
+            return await this.getTraceByPoints(program,ion,material,plot_using)
+    }
+
+    async getTraceByPoints(program, ion, material, points) {
+        const wasm = await this.wasm()
+        const start = wasm.ccall('dedx_get_min_energy', 'number', ['number','number'],[program,ion])
+        const end = wasm.ccall('dedx_get_max_energy', 'number', ['number','number'],[program,ion])
+
+        const xs = TraceFactory.getXAxisByPoints(start,end,points)
+
+        const stepFunction = wasm.cwrap('dedx_get_simple_stp','number',['number', 'number', 'number', 'number'])
```

Comment:
Are this functions already implemented and merged into mail libdedx ?

Reply:
Yes, it's one of the original functions

Reply:
It will still be changed after we introduce the new function working on whole arrays of data

Reply:
ok, got it